### PR TITLE
Update release notes to include preview feature section

### PR DESCRIPTION
**Pull request by lkasso**
### Details
This is a Documentation change ONLY.

### Context
Release Notes now have a "Experimental Features" section for Gradle features in alpha/beta/testing mode.
The Feature Lifecycle has been updated accordingly.

### CODE REVIEW

File: platforms/documentation/docs/src/docs/release/notes-template.md [hunk 1]
```diff
@@ -71,6 +71,15 @@ The following are the features that have been promoted in this Gradle release.
 ### Example promoted
 -->
 
+## Preview features
```
Comment (big-guy) [inline platforms/documentation/docs/src/docs/release/notes-template.md:74]: I think preview features is maybe the wrong phrase here. We already have something called "feature previews". Feature previews are a way we introduce breaking behavior and allow builds to opt into the new behavior early.

e.g., type-safe project accessors are both an incubating feature and enabled with a feature preview.  Not every incubating feature will have a feature preview and not every feature preview will align with an incubating feature. 

A preview feature isn't a part of our feature lifecycle (internal -> incubating -> public -> deprecated).  We have the "promoted features" section above which is supposed to signal something going from incubating -> public/stable.  This usually happens after a long period of no other changes, so there's nothing significant to report. 

I'm not sure if we really want/need an entirely separate section dedicated to incubating (or preview features). That would imply the other section can only be updates to existing stable features.  Maybe we just need to be clear when an update is about an incubating feature/API or use some kind of special header decoration, like we do in the DSL reference.

**Review by big-guy — CHANGES_REQUESTED** [review]
I think I'm -1 to this.  I'm not really sure we want to split the release notes into stable/unstable update sections.

File: platforms/documentation/docs/src/docs/userguide/releases/feature_lifecycle.adoc [hunk 2]
```diff
@@ -49,8 +50,24 @@ If it appears in this User Manual, the DSL Reference, or the API Reference, then
...
+== 2. Experimental
+
+_Experimental_ features are in an early-stage preview, they have been introduced and tested in Gradle; however, full usability and edge-case handling may not be complete yet.
+
+An _experimental_ feature gives you a chance to try out the feature while it's still in development.
+_Experimental_ features will (generally) not be documented.
+
+_Experimental_ features are not enabled by default.
+
+The feature might significantly change or be removed at any time.
```
Comment (oleg-nenashev) [inline platforms/documentation/docs/src/docs/userguide/releases/feature_lifecycle.adoc:63]:
```suggestion
The experimental feature might significantly change or be removed at any time, without prior notice.
```

File: platforms/documentation/docs/src/docs/userguide/releases/feature_lifecycle.adoc [hunk 3]
```diff
@@ -49,8 +50,24 @@ If it appears in this User Manual, the DSL Reference, or the API Reference, then
...
+We do not guarantee the use of these features against defects that may produce unexpected or undesired results.
```
Comment (oleg-nenashev) [inline platforms/documentation/docs/src/docs/userguide/releases/feature_lifecycle.adoc:64]:
```suggestion
We do not guarantee the use of these features against defects that may produce unexpected or undesired results,
including but not limited to potential security risks.
```

File: platforms/documentation/docs/src/docs/userguide/releases/feature_lifecycle.adoc [hunk 4]
```diff
@@ -49,8 +50,24 @@ If it appears in this User Manual, the DSL Reference, or the API Reference, then
...
+_Experimental_ features should not be used in production systems or with production data.
+
+Eventually _experimental_ features either graduate to _incubating_ or are removed from Gradle.
```
Comment (oleg-nenashev) [inline platforms/documentation/docs/src/docs/userguide/releases/feature_lifecycle.adoc:67]: "Graduate" seems to be a stronger word

```suggestion
Eventually _experimental_ features either get promoted to _incubating_ or are removed from Gradle.
```

File: platforms/documentation/docs/src/docs/userguide/releases/feature_lifecycle.adoc [hunk 5]
```diff
@@ -49,8 +50,24 @@ If it appears in this User Manual, the DSL Reference, or the API Reference, then
...
+
 [[sec:incubating_state]]
-== 2. Incubating
+== 3. Incubating
 
 Features are introduced in the _incubating_ state to allow real-world feedback to be incorporated into the feature before making it public.
```
Comment (oleg-nenashev) [inline platforms/documentation/docs/src/docs/userguide/releases/feature_lifecycle.adoc:72]:
```suggestion
The _incubating_ state for new features is designed to allow real-world feedback to be incorporated into the feature before making it public.
```

File: platforms/documentation/docs/src/docs/userguide/releases/feature_lifecycle.adoc [hunk 6]
```diff
@@ -49,8 +50,24 @@ If it appears in this User Manual, the DSL Reference, or the API Reference, then
...
 
+[[sec:experimental_state]]
+== 2. Experimental
+
+_Experimental_ features are in an early-stage preview, they have been introduced and tested in Gradle; however, full usability and edge-case handling may not be complete yet.
+
+An _experimental_ feature gives you a chance to try out the feature while it's still in development.
+_Experimental_ features will (generally) not be documented.
+
+_Experimental_ features are not enabled by default.
```
Comment (oleg-nenashev) [inline platforms/documentation/docs/src/docs/userguide/releases/feature_lifecycle.adoc:61]:
```suggestion
_Experimental_ features are not enabled by default, and enabling them might require multiple steps and a special environment.
Follow the feature documentation.
```

**Review by oleg-nenashev — COMMENTED** [review]
Looks good overall, just clarufying some bits

**Comment by lkasso** [comment]
Adding an "Experiment" feature lifecycle is no longer needed to improve and produce guidelines for release notes.

**Comment by oleg-nenashev** [comment]
Would be nice to summarize the decision for the community members

**Pull request closed without merging**